### PR TITLE
Habana DLC 1.3.0 Release (derived from PR #1692)

### PR DESCRIPTION
**Pull request by shantanutrip**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
DLC bot failed to funciton properly on PR #1692 that was created by the Habana team originally. This PR is built on top of PR #1692 to fix the leftover bugs.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by omrialmog — COMMENTED** [review]
Can we please rename the change to "Habana DLC 1.3.0 Release"?

File: test/test_utils/__init__.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
 # Since latest driver is not in public DLAMI yet, using a custom one
 NEURON_UBUNTU_18_BASE_DLAMI_US_WEST_2 = "ami-078c2404eecfbe916"
 # Habana Base v1.2 ami
-UBUNTU_18_HPU_DLAMI_US_WEST_2 = "ami-047fd74c001116366"
+UBUNTU_18_HPU_DLAMI_US_WEST_2 = "ami-0ef18b1906e7010fb" 
 UBUNTU_18_HPU_DLAMI_US_EAST_1 = "ami-04c47cb3d4fdaa874"
```
[the remark this answers is not in the document]
Comment (shantanutrip) [reply test/test_utils/__init__.py]: Done, thanks

File: dlc_developer_config.toml [hunk 2]
```diff
@@ -12,7 +12,7 @@ neuron_mode = false
 graviton_mode = false
 # Please only set it to True if you are preparing a HABANA related PR
 # Do remember to revert it back to False before merging any PR (including HABANA dedicated PR)
-habana_mode = false
+habana_mode = true
```
[the remark this answers is not in the document]
Comment (shantanutrip) [reply dlc_developer_config.toml:15]: Yes, I will revert it, thanks.

File: test/testrunner.py [hunk 3]
```diff
@@ -352,7 +352,7 @@ def main():
                 context.run("git clone https://github.com/HabanaAI/gaudi-test-suite.git")
```
Comment (shantanutrip) [inline test/testrunner.py:352]: Will revert

File: habana/tensorflow/training/docker/2.8/py3/synapseai1.3.0/Dockerfile.hpu [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    libsqlite3-dev \
+    libssl-dev \
+    tk-dev \
+ && rm -rf /var/lib/apt/lists/* \
+ && apt-get clean
+
+
+RUN ${PIP} --no-cache-dir install --upgrade \
+    pip \
+    #Setuptools release 60.x (and later) revealed an issue in Habana-TensorFlow package.
```
Comment (tejaschumbalkar) [inline habana/tensorflow/training/docker/2.8/py3/synapseai1.3.0/Dockerfile.hpu:83]: is this still valid @omrialmog ?

File: test/dlc_tests/sanity/test_pre_release.py [hunk 5]
```diff
@@ -194,6 +194,12 @@ def test_framework_version_cpu(image):
         else:
             if "neuron" in image:
                 assert tag_framework_version in output.stdout.strip()
+            if all(_string in image for _string in ["pytorch", "habana", "synapseai1.3.0"]):
+                # Habana Pytorch version looks like 1.10.0a0+gitb488e78 for SynapseAI1.3 PT1.10.1 images
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/test_pre_release.py:198]: is this intentional?

File: test/test_utils/__init__.py [hunk 6]
```diff
@@ -43,8 +43,8 @@
 # Since latest driver is not in public DLAMI yet, using a custom one
 NEURON_UBUNTU_18_BASE_DLAMI_US_WEST_2 = "ami-078c2404eecfbe916"
 # Habana Base v1.2 ami
```
Comment (tejaschumbalkar) [inline test/test_utils/__init__.py:45]: nit: v1.3 ami

File: habana/tensorflow/training/docker/2.8/py3/synapseai1.3.0/Dockerfile.hpu (hunk 4 above)
Comment (omrialmog) [reply habana/tensorflow/training/docker/2.8/py3/synapseai1.3.0/Dockerfile.hpu:83]: @shantanutrip had run all the testing with this in place. I dont believe its required anymore. I would recommend maybe removing this in the next release so you dont have to run another full test cycle for this release.

File: test/dlc_tests/sanity/test_pre_release.py (hunk 5 above)
Comment (shantanutrip) [reply test/dlc_tests/sanity/test_pre_release.py:198]: Yes, I think for PT1.10.1 images, the PT version seen within the image is `1.10.0a0+gitb488e78`. I think this is from the PT side (https://github.com/pytorch/pytorch/blob/v1.10.1/version.txt)

File: test/test_utils/__init__.py (hunk 6 above)
Comment (shantanutrip) [reply test/test_utils/__init__.py:45]: Will do, thanks.

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**